lib/net.js

```diff
@@ -497,7 +498,7 @@ Socket.prototype.destroySoon = function() {
 Socket.prototype._destroy = function(exception, cb) {
   debug('destroy');
 
-  function fireErrorCallbacks(self) {
+  function fireErrorCallbacks(self, exception, cb) {
```

Comment:
You mean shadowing of variable/argument names?  No, should not be a factor here.